bigframes/core/blocks.py

```diff
@@ -2011,6 +2012,61 @@ def _is_monotonic(
         self._stats_cache[column_name].update({op_name: result})
         return result
 
+    def _get_rows_as_json_values(self) -> Block:
+        sql, index_column_ids, _ = self.to_sql_query(include_index=True)
+
+        # all column names, types and values
+        all_column_names = index_column_ids + [col for col in self.column_labels]
+        column_names_csv = ", ".join([f'"{col}"' for col in all_column_names])
+        column_references_csv = ", ".join(
+            [f"CAST({col} AS STRING)" for col in all_column_names]
+        )
```

Comment:
this will error in bigquery for some types, lets make our type constraints clear

Reply:
maybe TO_JSON_STRING is a bit more flexible?

Reply:
I made the column type handling more generic, PTAL.